src/main/java/org/candlepin/resource/RootResource.java

```diff
@@ -75,6 +77,33 @@
         addResource(RootResource.class);
         addResource(DistributorVersionResource.class);
         addResource(DeletedConsumerResource.class);
+
+        // Due to OpenAPI changes, endpoints to access content override & guestIDs
+        // were moved to ConsumerResource class. Here URL links for content override &
+        // guestIDs are accessed via methods specific to content override & guestIDs
+        Class<?>[] implementedInterfaces = ConsumerResource.class.getInterfaces();
+        Class<?> consumersAPI = null;
+
+        for (Class<?> apiClass: implementedInterfaces) {
+            if (apiClass.getSimpleName().equals("ConsumersApi")) {
```

Comment:
While this gets the job done, I'm not at all thrilled about having the names of both the interface and the methods hardcoded into this obscure class. If we need to start manually flagging resources for entry into the list of links, we should be doing something like annotating the methods in question (or flagging them in the API spec to get annotated by the code generator) and then picking that up automatically.

This is just priming us to run into the same bug during a simple refactor and hide it in a one-off error that only shows up the first time the endpoint is hit, and then never again.

I've created a PoC of this which you may use however you like: https://gist.github.com/Ceiu/b6694cba3b5c619987aa4a324a068914

Reply:
Really liked the idea to use annotations here. Thanks.